src/datastore/types.ts

```diff
@@ -1,10 +1,15 @@
 import { ICustomType } from "../types/types.ts";
 import { ManifestDatastoreSchema } from "../manifest/manifest_schema.ts";
 import { SlackManifest } from "../manifest/mod.ts";
+import type { ValidSchemaTypes } from "../schema/schema_types.ts";
+import type { ValidSlackPrimitiveTypes } from "../schema/slack/types/mod.ts";
+import type { LooseStringAutocomplete } from "../type_utils.ts";
 
 export type SlackDatastoreAttribute = {
   // supports custom types, primitive types, inline objects and lists
-  type: string | ICustomType;
+  type:
+    | LooseStringAutocomplete<ValidSchemaTypes | ValidSlackPrimitiveTypes>
+    | ICustomType;
```

Comment:
@chris-skud I wanted to confirm with you that all of these types are valid for Datastores:

```ts
"string",
"boolean",
"integer",
"number",
"object",
"array",
"slack#/types/user_id",
"slack#/types/channel_id",
"slack#/types/usergroup_id",
"slack#/types/date",
"slack#/types/timestamp",
"slack#/types/blocks",
"slack#/types/credential/oauth2",
"slack#/types/rich_text",
"slack#/types/message_ts",
```

Reply:
Hmm, these aren't:
slack#/types/message_ts
slack#/types/blocks
slack#/types/credential/oauth2

I think `message_ts` should be and it's easy to add (a string if i'm not mistaken.) I'm not sure blocks and oauth2 would make sense.

Reply:
I'm prepping a PR for message_ts now.

Reply:
Awesome, thank you @chris-skud! This has been fixed and the new computed type looks like

```ts
type ValidDatastoreTypes =
  | "string"
  | "number"
  | "boolean"
  | "object"
  | "integer"
  | "array"
  | "slack#/types/user_id"
  | "slack#/types/channel_id"
  | "slack#/types/usergroup_id"
  | "slack#/types/date"
  | "slack#/types/timestamp"
  | "slack#/types/rich_text";
```

When support for `message_ts` goes in, we can easily remove it from the deny list

Reply:
@shapirone, server-side support for `message_ts` just now merged.

Reply:
✅